src/scripting/LuaSandbox.cpp

```diff
@@ -478,7 +488,7 @@ void LuaSandbox::InitializeIOForSandbox(Sandbox& aSandbox, const sol::state& acp
         return res;
     };
 
-    sbEnv["dir"] = [cSBRootPath, stateView](const std::string& acPath) -> sol::table
+    sbEnv["dir"] = [this, cSBRootPath, stateView](const std::string& acPath) -> sol::table
```

Comment:
passing this into capture is problematic, sandbox pointer is not guaranteedto be stable during whole runtime, that is also why we pass LuaSandbox along with index around.

Reply:
you have this in few places